spec/features/consumer/authentication_spec.rb

```diff
@@ -135,5 +136,39 @@
       uri = URI.parse(current_url)
       expect(uri.path + "#" + uri.fragment).to eq('/#/login')
     end
+
+    describe "with user locales" do
+      before do
+        visit root_path
+        open_login_modal
+      end
+
+      context "when the user has a valid locale saved" do
+        before do
+          user.update!(locale: "es")
+        end
+
+        it "logs in successfully, applying the saved locale" do
+          fill_in_and_submit_login_form(user)
+          expect_logged_in
+
+          expect(page).to have_content I18n.t(:home_shop, locale: :es).upcase
+        end
+      end
+
+      context "when the user has an unavailable locale saved" do
+        before do
+          user.update!(locale: "xx")
+        end
+
+        it "logs in successfully and resets the user's locale to the default" do
+          fill_in_and_submit_login_form(user)
+          expect_logged_in
+
+          expect(page).to have_content I18n.t(:home_shop, locale: :en).upcase
+          expect(user.reload.locale).to eq "en"
+        end
+      end
+    end
```

Comment:
I don't really see why we need this to be feature specs instead of controller. This logic only affects that controller and I'm pretty sure you can stub the cookie from controller specs. I'm worried about the pace at which increase the build's execution time

Reply:
I tried it in a couple of different lower-down specs first, but it didn't replicate the error, everything was passing. I managed to replicate it in a feature spec though. I think this case involves some "automagic" things going on between Spree, Rails and Devise, but I'm not sure exactly.

Reply:
In the stackstace, the only lines of code that were actually in our codebase was one method in `I18nHelper`.

Reply:
I've added more unit tests here as well, but a couple of small integration tests that tell us whether or not users can actually log in to the app without hitting a fatal error seem like a good idea. 

Valen la pena, no? :sweat_smile:

Reply:
I wonder what kind of tests in this area we already have. We have an authentication_spec. There's also a login in registration_spec. I wonder if we could add some locale variant in one of these login specs.

Reply:
Already added in authentication_spec :+1:

Reply:
Fair enough. I was convinced we login through the UI in several feature specs but it doesn't look like it now.